q2_feature_classifier/_consensus_assignment.py

```diff
@@ -116,8 +116,13 @@ def _blast6format_df_to_series_of_lists(
     # accession ID, so we will add this mapping to the reference taxonomy.
     ref_taxa['*'] = unassignable_label
     # map accession IDs to taxonomy
-    taxa_hits.replace(ref_taxa, inplace=True)
+    for index, value in assignments.iterrows():
+        sseqid = assignments.iloc[index]['sseqid']
+        assignments.at[index, 'sseqid'] = ref_taxa.at[sseqid]
```

Comment:
This will ultimately cause `search_results` in `find_consensus_annotation` to be mutated.

That's probably fine, but something to keep track of. 
Would it be too much to do a df.copy() first? 
(I realize that memory pressure is the entire purpose of this change, so maybe it would be too much.)

Reply:
Otherwise, as best I can tell. This does the same thing as before. So once our tests run, we should see the same results.

Reply:
Good idea! It should be possible to use a `df.copy()` copy at a minimal cost. The problem with `replace` is that somehow it was using exponential memory (a reported issue for this pandas version)